jdbc/src/main/java/org/apache/metamodel/jdbc/JdbcDataSet.java

```diff
@@ -197,8 +197,12 @@ public void close() {
         if (_closed) {
             return;
         }
+
+        FileHelper.safeClose(_resultSet);
+
         if (_jdbcDataContext != null) {
-            _jdbcDataContext.close(_connection, _resultSet, _statement);
+            FileHelper.safeClose(_statement);
```

Comment:
Ah. I think I know: The statement for compiled queries can't be closed with the dataset, then next execution will fail.